contracts/vaults/STETHVault/RibbonThetaSTETHVault.sol

```diff
@@ -0,0 +1,353 @@
+        address _marginPool,
+        address _gnosisEasyAuction,
+        address _crvPool
+    )
+        RibbonVault(
+            _weth,
+            _usdc,
+            _ldo,
+            _gammaController,
+            _marginPool,
+            _gnosisEasyAuction,
+            _crvPool
+        )
+    {
+        require(_oTokenFactory != address(0), "!_oTokenFactory");
+        OTOKEN_FACTORY = _oTokenFactory;
+    }
+
+    /**
+     * @notice Initializes the OptionVault contract with storage variables.
+     */
+    function initialize(
+        address _owner,
+        address _keeper,
+        address _feeRecipient,
+        uint256 _managementFee,
+        uint256 _performanceFee,
+        string memory tokenName,
+        string memory tokenSymbol,
+        address _optionsPremiumPricer,
+        address _strikeSelection,
+        uint32 _premiumDiscount,
+        uint256 _auctionDuration,
+        Vault.VaultParams calldata _vaultParams
+    ) external initializer {
+        baseInitialize(
+            _owner,
+            _keeper,
+            _feeRecipient,
+            _managementFee,
+            _performanceFee,
+            tokenName,
+            tokenSymbol,
+            _vaultParams
+        );
+        require(_optionsPremiumPricer != address(0), "!_optionsPremiumPricer");
+        require(_strikeSelection != address(0), "!_strikeSelection");
+        require(
+            _premiumDiscount > 0 && _premiumDiscount < 1000,
+            "!_premiumDiscount"
+        );
+        require(_auctionDuration >= 1 hours, "!_auctionDuration");
+        optionsPremiumPricer = _optionsPremiumPricer;
+        strikeSelection = _strikeSelection;
+        premiumDiscount = _premiumDiscount;
+        auctionDuration = _auctionDuration;
+    }
+
+    /************************************************
+     *  SETTERS
+     ***********************************************/
+
+    /**
+     * @notice Sets the new discount on premiums for options we are selling
+     * @param newPremiumDiscount is the premium discount
+     */
+    function setPremiumDiscount(uint16 newPremiumDiscount) external onlyOwner {
+        require(
+            newPremiumDiscount > 0 && newPremiumDiscount < 1000,
+            "Invalid discount"
+        );
+
+        premiumDiscount = newPremiumDiscount;
+    }
+
+    /**
+     * @notice Sets the new auction duration
+     * @param newAuctionDuration is the auction duration
+     */
+    function setAuctionDuration(uint256 newAuctionDuration) external onlyOwner {
+        require(newAuctionDuration >= 1 hours, "!newAuctionDuration");
+
+        auctionDuration = newAuctionDuration;
+    }
+
+    /**
+     * @notice Sets the new strike selection or options premium pricer contract
+     * @param newContract is the address of the new strike selection or options premium pricer contract
+     * @param isStrikeSelection is whether we are setting the strike selection contract
+     */
+    function setStrikeSelectionOrPricer(
```

Comment:
condensed setStrikeSelection / setOptionsPremiumPricer to one method to fit contract size constraint